### ISSUE TITLE
SEB doesn't create a taskbar for a 2nd display screen

### ISSUE THREAD

**Issue by Notselwyn** [SafeExamBrowser/seb-win-refactoring#564]
**Describe the Bug**
SafeExamBrowser for Windows doesn't create a taskbar for a second screen when using 2+ screens.

**Steps to Reproduce**
1. Set maximum allowed number of connect displays in the SEB config to atleast 2
2. Run SEB with 2 or more display screens

**Expected Behavior**
SEB should create a taskbar on all screens, like Windows: 
![image](https://user-images.githubusercontent.com/68616630/218266557-8b07856f-22c7-4a03-a5ca-2ee9ec32e705.png)

**Screenshots**
(Disabled kiosk mode for taking the screenshot: the bug persists with kiosk mode enabled, but it won't show the Windows taskbar either - which is good)
![image](https://user-images.githubusercontent.com/68616630/218266757-9a23b1e5-0885-4d04-873b-09750370230d.png)

**Version Information**
 - OS: Windows 10 Pro 22H2
 - SEB-Version: 3.5.0-beta (earlier versions are effected as well)

**Comment by danschlet** [comment]
The macOS version also only displays the taskbar/dock on the main display, although if several displays are allowed in settings, the main browser window (the one where the Start URL was loaded and the exam is displayed in) can be moved to another screen, then the SEB Dock/taskbar is moved to that screen as well. It is therefore always displayed only on the screen, where the main SEB browser window is. This is identical to the macOS Dock behavior with standard settings.

**Comment by heinzkrebs** [comment]
I would like to see this feature as well!

We are currently planning an enviroment with connected Wacom-Displays, and thus i need to span the browser across both displays. 

ciao
  H.

**Comment by danschlet** [comment]
What exactly is it what you want to achieve? You should be able to open a second window in SEB when a web application or website uses a link which opens in another browser window. That second window you can move on another display, as long as the SEB settings you're using allow multiple displays.

**Comment by heinzkrebs** [comment]
my application is just ONE HTML page, which is divided into 2 DIVs, the
left should be displayed on monitor 1 , the right on monitor 2

so i need to span the browser over both displays

i can do this manually using firefox and drag an drop the borders, but
this is not the good way ...

sincerely
  Heinz

Am Freitag, dem 16.06.2023 um 05:10 -0700 schrieb Daniel R. Schneider:

**Comment by danschlet** [comment]
Ah, on Windows, (browser) windows can span multiple displays? On Mac that's not possible, a window can only be displayed on one screen. Because we prioritize SEB features which work on all platforms, support for this likely won't have a high priority (I think @dbuechel will agree with this).

Maybe you could solve this issue for all platforms if you implement an option to split your one html page to two windows? Then it at least would be better usable also on Macs?

I don't know your app, but it seems weird to span a window over two displays especially when one of them is a wacom tablet display, which usually lies flat on the desk and doesn't stand vertically like a regular monitor. How can people make sure that one div is on the vertical display and the second one on the tablet? I guess you anyways have to provide so scalabiltiy for your content/divs, so IMHO it doesn't seem impossible to split it up to two windows?

**Comment by heinzkrebs** [comment]
hello,

> Ah, on Windows, (browser) windows can span multiple displays? On Mac
> that's not possible, a window can only be displayed on one screen.

also possible for mac, if i understand that correctly:
https://www.duetdisplay.com/help-center/how-do-i-have-a-window-span-multiple-monitors-on-macos

> Because we prioritize SEB features which work on all platforms,
> support for this likely won't have a high priority (I think @dbuechel
> will agree with this).

i understand ..
but from my understanding i can find several configuration options in
the config-editor, which are explicit for windows and others for mac?!?

> Maybe you could solve this issue for all platforms if you implement
> an option to split your one html page to two windows? Then it at
> least would be better usable also on Macs?

but how do i guarantee, that the new window/html page is opened on the
second display, and not on the first with SEB???

> I don't know your app, but it seems weird to span a window over two
> displays 

in a gaming "society" this is quite common (but of course this is not
my target :)) )

> especially when one of them is a wacom tablet display, which usually
> lies flat on the desk and doesn't stand vertically like a regular
> monitor. How can people make sure that one div is on the vertical
> display and the second one on the tablet? 

the displays are "ordered", so the monitor is always display 1 and the
wacom is always display 2 - it's a controlled environment.

i know that this is not a perfect solution, but currently this is the
only option i have. it's a nogo to tell the students to press shortkey
Alt-XYZ and then move a window, and, and - before starting an exam. and
i have less possibilities to confirgure the environment, since it is
VDI-environment ...

and so i thought that the original post is quite similar to my
requests, thus i "hijacked" the post ...

ciao
  H.

**Comment by dbuechel** [comment]
As @danschlet said, this particular feature request as well as the taskbar on both screens resp. multi-screen support currently unfortunately has a very low priority, as we are completely booked out with other new features and functionalities. But we'll definitely keep this open so that we may get back to it once there is more capacity available again.
